### PR TITLE
Handle empty descriptions in shops

### PR DESCRIPTION
**Pull request by warp-core**
**Bugfix:**

## Fix Details
Currently, the description click zone and collapsed label will still be created and drawn even if the description is completely empty.
This PR skips these items if the description is empty.

## Testing Done
Visited the outfitter and shipyard and checked items with no description:
Before | After
-- | --
image | image
image |

### CODE REVIEW

File: source/ItemInfoDisplay.cpp [hunk 1]
```diff
@@ -162,7 +162,10 @@ void ItemInfoDisplay::UpdateDescription(const string &text, const vector<string>
 	}
 
 	// Pad by 10 pixels on the top and bottom.
-	descriptionHeight = description.Height() + 20;
+	// Unless there is no description.
```
Comment (RisingLeaf) [inline source/ItemInfoDisplay.cpp:165]: Suggestion:
If there is a description, pad by 10 pixels on the top and bottom.

**Review by RisingLeaf — APPROVED** [review]

File: source/ItemInfoDisplay.cpp [hunk 2]
```diff
@@ -162,7 +162,10 @@ void ItemInfoDisplay::UpdateDescription(const string &text, const vector<string>
 	}
 
 	// Pad by 10 pixels on the top and bottom.
```
Comment (Amazinite) [inline source/ItemInfoDisplay.cpp:164]:
```suggestion
	// If there is a description, pad by 10 pixels on the top and bottom.
```

File: source/ItemInfoDisplay.cpp (hunk 1 above)
Comment (Amazinite) [inline source/ItemInfoDisplay.cpp:165]:
```suggestion
```

File: source/OutfitterPanel.cpp [hunk 3]
```diff
@@ -266,28 +266,33 @@ int OutfitterPanel::DrawDetails(const Point &center)
...
+				SpriteShader::Draw(collapsedArrow, startPoint + Point(20., 20.));
+			}
 
-		// Calculate the ClickZone for the description and add it.
-		const Point descriptionDimensions(INFOBAR_WIDTH, descriptionOffset);
-		const Point descriptionCenter(center.X(), startPoint.Y() + descriptionOffset / 2);
-		ClickZone<string> collapseDescription = ClickZone<string>(
-			descriptionCenter, descriptionDimensions, DESCRIPTION);
-		categoryZones.emplace_back(collapseDescription);
+			// Calculate the new ClickZone for the description.
```
Comment (Amazinite) [inline source/OutfitterPanel.cpp:289]:
```suggestion
			// Calculate the ClickZone for the description and add it.
```

File: source/ShipyardPanel.cpp [hunk 4]
```diff
@@ -179,28 +179,33 @@ int ShipyardPanel::DrawDetails(const Point &center)
...
+				SpriteShader::Draw(collapsedArrow, startPoint + Point(20., 20.));
+			}
 
-		// Calculate the ClickZone for the description and add it.
-		const Point descriptionDimensions(INFOBAR_WIDTH, descriptionOffset);
-		const Point descriptionCenter(center.X(), startPoint.Y() + descriptionOffset / 2);
-		const ClickZone<string> collapseDescription = ClickZone<string>(
-			descriptionCenter, descriptionDimensions, DESCRIPTION);
-		categoryZones.emplace_back(collapseDescription);
+			// Calculate the new ClickZone for the description.
```
Comment (Amazinite) [inline source/ShipyardPanel.cpp:202]:
```suggestion
			// Calculate the ClickZone for the description and add it.
```

**Pull request merged**